### ISSUE TITLE
Add sniff to delete_option() + add_option

### ISSUE THREAD

**Issue by GaryJones** [Automattic/VIP-Coding-Standards#637]
## What problem would the enhancement address for VIP?
As explained at https://github.com/woocommerce/woocommerce/pull/27696 there is a race condition when `delete_option('foo')` is followed by `add_option('foo', '...')`.

## Describe the solution you'd like

Having a sniff that looked for a `delete_option()` call followed by an `add_option()` immediately 
or at some point within the same scope, for the same option key, could flag this race condition and suggest using `update_option()` instead.

## What code should be reported as a violation?

```php
function reset_settings() {
	$defaults = get_default_settings();
	delete_option( 'my_settings' );
	add_option( 'my_settings', $defaults );
}
```

Likely to need some consideration of an option key that is using a variable.

## What code should *not* be reported as a violation?

- `add_option()` in a different scope.
- different option key (including different variable).

**Comment by rebeccahum** [comment]
> or at some point within the same scope

Do you have any examples on what would be considered in scope?

**Comment by GaryJones** [comment]
Within the same function or method call.

**Comment by rebeccahum** [comment]
Starting to work on this one roughly...I believe that the `add_option()` call with the same option name would have to follow the `delete_option()`. E.g. the sniff would not flag this as a race condition:

```
function reset_settings() {
	$defaults = get_default_settings();
	delete_option( 'my_settings' );
	add_option( 'settings', [] );
	add_option( 'my_settings', $defaults );
}
```

**Comment by GaryJones** [comment]
Can certainly start with looking at the calls being in consecutive statements.
